cmd/crd-schema-gen/main.go

```diff
@@ -0,0 +1,292 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"reflect"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	crdgenerator "sigs.k8s.io/controller-tools/pkg/crd/generator"
+
+	_ "github.com/openshift/api"
+)
+
+var (
+	scheme = runtime.NewScheme()
+	codecs = serializer.NewCodecFactory(scheme)
+)
+
+func init() {
+	v1beta1.AddToScheme(scheme)
+}
+
+func main() {
+	if err := run(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+func run() error {
+	apisDir := flag.String("apis-dir", "pkg/apis", "the Golang API directory")
+	apis := flag.String("apis", "*", "the apis to generate from the apis-dir, in bash glob syntax")
+	manifestDir := flag.String("manifest-dir", "manifests", "the directory with existing CRD manifests")
+	outputDir := flag.String("output-dir", "", "optional directory to output the kubebuilder CRDs. By default a temporary directory is used.")
+	verifyOnly := flag.Bool("verify-only", false, "do not write files, only compare and return with return code 1 if dirty")
```

Comment:
sound like `dry-run` ?

Reply:
called verify-only elsewhere. It's more than dry-run, fails on differences.